File: Bai_86/main.cpp
#include <iostream>
using namespace std;
void count_way(int n){
    int i = 1;
    int way_count = 0;
    for(int ten = 0; ten <= n / 10; ++ten) {
        for (int five = 0; five <= (n - ten * 10) / 5; ++five) {
            int two = n - ten * 10 - five * 5;
            if (two % 2 == 0) {
                int one = two / 2;
                cout << i++ << ':' << ten << " to 10 " << five << " to 5 " << one << " to 2 " << endl;
                way_count++;
            }
        }
    }
    cout << "Tong so cach: " << way_count << endl;
}

int main() {
    int n;
    do{
        cout << "Nhap n: ";
        cin >> n;
        if(n < 0  || n > 50) cout << "Nhap lai n" << endl;
    } while (n < 0  || n > 50);
    count_way(n);
    return 0;
}

File: Bai_86/main.py
def count_ways(n):
    ways_count = 0
    for ten in range(n // 10 + 1):
        for five in range((n - ten * 10) // 5 + 1):
            two = n - ten * 10 - five * 5
            if two % 2 == 0:
                ways_count += 1
    return ways_count

N = int(input("Nhập số tiền cần đổi (N <= 50): "))
if N <= 50:
    print("Có tổng cộng", count_ways(N), "phương án.")
else:
    print("Số tiền cần đổi không được lớn hơn 50.")




def print_ways(n):
    i = 1
    for ten in range(n // 10 + 1):
        for five in range((n - ten * 10) // 5 + 1):
            two = n - ten * 10 - five * 5
            if two % 2 == 0:
                one = two // 2
                print(f"{i} : {ten} tờ 10, {five} tờ 5, {one} tờ 2")
                i += 1

N = int(input("Nhập số tiền cần đổi (N <= 50): "))
if N <= 50:
    print("Các phương án đổi tiền:")
    print_ways(N)
else:
    print("Số tiền cần đổi không được lớn hơn 50.")
